File: scripts/verify-product-images.py
import json
import os
import subprocess
import sys
from pathlib import Path

root = Path(__file__).resolve().parent.parent
catalog_path = root / 'content' / 'valenza-catalog.json'
with catalog_path.open('r', encoding='utf-8') as fh:
    catalog = json.load(fh)
products = catalog.get('products', [])
refs = []
for product in products:
    for key in ['image', 'imageThumb', 'imageMedium']:
        value = product.get(key)
        if isinstance(value, str) and value.startswith('/'):
            refs.append((product.get('slug'), key, value))
    images = product.get('images') or []
    if isinstance(images, list):
        for src in images:
            if isinstance(src, str) and src.startswith('/'):
                refs.append((product.get('slug'), 'images', src))

bad = []
for slug, key, src in refs:
    full_path = root / src.lstrip('/')
    exists = full_path.exists()
    ok = False
    if exists:
        ext = full_path.suffix.lower()
        ok = ext in {'.jpg', '.jpeg', '.png', '.webp', '.gif', '.avif', '.bmp'}
    if not exists or not ok:
        bad.append((slug, key, src, exists, full_path.suffix.lower() if full_path.exists() else ''))

print('total refs', len(refs))
print('bad count', len(bad))
for item in bad[:200]:
    print(item[0], item[1], item[2], 'exists=', item[3], 'ext=', item[4])

File: scripts/verify-product-images.mjs
import fs from 'fs';
import path from 'path';
import sharp from 'sharp';

const root = process.cwd();
const catalogPath = path.join(root, 'content', 'valenza-catalog.json');
const catalog = JSON.parse(fs.readFileSync(catalogPath, 'utf8'));
const products = catalog.products || [];
const refs = [];

for (const product of products) {
  for (const key of ['image', 'imageThumb', 'imageMedium']) {
    const value = product[key];
    if (typeof value === 'string' && value.startsWith('/')) {
      refs.push({ slug: product.slug, key, src: value });
    }
  }
  if (Array.isArray(product.images)) {
    for (const src of product.images) {
      if (typeof src === 'string' && src.startsWith('/')) {
        refs.push({ slug: product.slug, key: 'images', src });
      }
    }
  }
}

const bad = [];

for (const ref of refs) {
  const full = path.join(root, ref.src.replace(/^\//, ''));
  const exists = fs.existsSync(full);
  let ok = false;
  let err = '';
  if (exists) {
    try {
      const metadata = await sharp(full).metadata();
      ok = Boolean(metadata.format);
    } catch (error) {
      err = error.message;
    }
  }
  if (!exists || !ok) {
    bad.push({ slug: ref.slug, src: ref.src, exists, err });
  }
}

console.log('total refs', refs.length);
console.log('bad count', bad.length);
for (const item of bad.slice(0, 200)) {
  console.log(item.slug, item.src, 'exists=', item.exists, 'err=', item.err);
}
